src/components/AssignmentCreatorForm.jsx

```diff
@@ -1,79 +1,88 @@
 import React from 'react';
 import {t} from 'i18next';
 import classnames from 'classnames';
-import map from 'lodash-es/map';
 import chrono from 'chrono-node';
-
 import PropTypes from 'prop-types';
+import ImmutablePropTypes from 'react-immutable-proptypes';
 import {Field, reduxForm} from 'redux-form/immutable';
 
-import {dateToString} from '../util/timeFormatter';
-
-import renderField from './renderField';
-import renderSelectField from './renderSelectField';
+import AssignmentCreatorTextField from './AssignmentCreatorTextField';
+import AssignmentCreatorSelectField from './AssignmentCreatorSelectField';
 
 function validate(values) {
```

Comment:
Should the validation (as well as `parse` and  `format`) remain in the `AssignmentCreatorForm` or should this be extracted out somewhere?